src/librbd/io/ImageRequest.cc

```diff
@@ -463,6 +463,19 @@ void AbstractImageWriteRequest<I>::send_request() {
 
   aio_comp->set_request_count(object_extents.size());
   if (!object_extents.empty()) {
+    // reflect changes in object_extents back to m_image_extents
+    if (ret == 1) {
+      this->m_image_extents.clear();
+      for (auto& object_extent : object_extents) {
+        auto [extents, _] = io::util::object_to_area_extents(
+            &image_ctx, object_extent.object_no,
+            {{object_extent.offset, object_extent.length}});
+        for (auto extent : extents) {
+          this->m_image_extents.push_back(extent);
+        }
```

Comment:
Use `insert` method for this:
```suggestion
        this->m_image_extents.insert(this->m_image_extents.end(),
                                     image_extents.begin(),
                                     image_extents.end());
```

Reply:
Nice suggestion!